step-framework-parent/step-framework-timeseries/src/main/java/step/core/timeseries/metric/MetricTypeRenderSettings.java

```diff
@@ -0,0 +1,51 @@
+package step.core.timeseries.metric;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class contains all needed settings for rendering a visual chart.
+ */
+public class MetricTypeRenderSettings {
+    
+    private String groupingAttribute; // the default grouping
+    private List<String> groupingOptions; // in case we will allow the user to change the grouping like Google does.
```

Comment:
what is the exact meaning of this field + please remove references from comments

Reply:
groupingAttribute - explained above
groupingOptions - you said you want to follow the google approach. for each metric, you can have several options for grouping. If we don't want to allow the user to switch the grouping by, we can remove it.

Reply:
As discussed this list of fields also applies to filtering and corresponds to the list attributes supported by the metric (in addition to the custom ones)